Improves 'patterns-and-guards' documentation page

Qqwy opened the pull request:
While taking a deep-dive in the Elixir and Erlang documentation of guard tests for my upcoming presentation on guards, I found a small couple of inconsistencies that this PR clears up.

Conference Driven Development! :upside_down_face: 

---

This PR improves the 'patterns-and-guards' documentation page to:

- Include `receive` in the list of constructs that accept guards
- Fix in-doc links to the mentioned functions.

josevalim commented:
@wojtekmach maybe we should add auto linking of Kernel functions back?

wojtekmach commented:
@josevalim yeah, I'll take care of it.

Qqwy commented:
Should I in that case remove the manual linking of the Kernel functions from this PR, or leave it as-is?

josevalim commented on lib/elixir/pages/patterns-and-guards.md:
```diff
@@ -382,13 +382,13 @@ Check.empty?({})
 In the examples above, we have used the match operator (`=`) and function clauses to showcase patterns and guards respectively. Here is the list of the built-in constructs in Elixir that support patterns and guards.
 
-  * the match operator (`=`) (exceptionally does not support guards):
+  * the match operator (`=`) intentionally does _not_ support guards:
```
I think the word intentional is a bit weird here. I believe the word exception makes sense, because we just said that these are the constructs that support them. Alternatively we can move this particular note to the bottom.

Qqwy replied:
The disadvantage of using the word 'exceptionally' is that it makes it look like it might have to do with exceptions.

I agree that it makes sense to move this note down to the bottom.

josevalim commented:
> Should I in that case remove the manual linking of the Kernel functions from this PR, or leave it as-is?

Yes, please remove them.

Qqwy commented:
I have

- [x] Removed the manual linking to functions in `Kernel` and `Kernel.SpecialForms`
- [x] Moved the note about `=` not supporting guards to the end and reworded it.

josevalim commented on lib/elixir/pages/patterns-and-guards.md:
```diff
@@ -430,7 +425,16 @@ In the examples above, we have used the match operator (`=`) and function clause
   * custom guards can also be defined with `defguard/1` and `defguardp/1`. A custom guard can only be defined based on existing guards.
+  
+  
+Finally, the the match operator (`=`) does _not_ support guards:
```
```suggestion
Note that the match operator (`=`) does *not* support guards:
```

josevalim commented:
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:
